tests/integrations/langchain/langchain_nv_ai_endpoints_test.py

```diff
@@ -0,0 +1,543 @@
+    assert usage["requests"] == 1
+    assert usage["completion_tokens"] == 24
+    assert usage["prompt_tokens"] == 12
+    assert usage["total_tokens"] == 36
+
+    inputs = call.inputs
+    assert inputs["model"] == model
+    assert inputs["messages"] == [{"role": "user", "content": "Hello!"}]
+    assert inputs["max_tokens"] == 64
+    assert inputs["temperature"] == 0.0
+    assert inputs["top_p"] == 1
+
+
+@pytest.mark.skip_clickhouse_client  # TODO:VCR recording does not seem to allow us to make requests to the clickhouse db in non-recording mode
+@pytest.mark.vcr(
+    filter_headers=["authorization"],
+    allowed_hosts=["api.wandb.ai", "localhost"],
+)
+def test_chatnvidia_stream_quickstart(
+    client: weave.trace.weave_client.WeaveClient,
+) -> None:
+    api_key = os.environ.get("NVIDIA_API_KEY", "DUMMY_API_KEY")
+
+    nvidia_client = ChatNVIDIA(
+        api_key=api_key, model=model, temperature=0.0, max_tokens=64, top_p=1
+    )
+
+    response = nvidia_client.stream("Hello!")
+    answer = AIMessageChunk(content="")
+    for chunk in response:
+        answer += chunk
+        answer.usage_metadata = chunk.usage_metadata
+
+    calls = list(client.calls())
+    # need to make 2 because of the langchain integration getting a call in there
+    assert len(calls) == 2
+    call = calls[1]
+
+    assert answer.content is not None
+
+    assert op_name_from_ref(call.op_name) == "langchain.Llm.ChatNVIDIA-stream"
+    assert call.started_at is not None
+    assert call.started_at < call.ended_at  # type: ignore
+
+    output = call.output
+    assert output["model"] == model
+    assert output["object"] == "chat.completion"
+
+    print(call.summary["usage"][output["model"]])
+    usage = call.summary["usage"][output["model"]]  # type: ignore
+    assert usage["requests"] == 1
+    assert usage["completion_tokens"] == 24
+    assert usage["prompt_tokens"] == 12
+    assert usage["total_tokens"] == 36
+
+    inputs = call.inputs
+    assert inputs["model"] == model
+    assert inputs["messages"] == [{"role": "user", "content": "Hello!"}]
+    assert inputs["max_tokens"] == 64
+    assert inputs["temperature"] == 0.0
+    assert inputs["top_p"] == 1
+
+
+@pytest.mark.skip_clickhouse_client  # TODO:VCR recording does not seem to allow us to make requests to the clickhouse db in non-recording mode
+@pytest.mark.vcr(
+    filter_headers=["authorization"], allowed_hosts=["api.wandb.ai", "localhost"]
+)
+@pytest.mark.asyncio
+async def test_chatnvidia_async_stream_quickstart(
+    client: weave.trace.weave_client.WeaveClient,
+) -> None:
+    api_key = os.environ.get("NVIDIA_API_KEY", "DUMMY_API_KEY")
+
+    nvidia_client = ChatNVIDIA(
+        api_key=api_key, model=model, temperature=0.0, max_tokens=64, top_p=1
+    )
+    response = nvidia_client.astream("Hello!")
+    answer = AIMessageChunk(content="")
+    async for chunk in response:
+        answer += chunk
+        answer.usage_metadata = chunk.usage_metadata
```

Comment:
Same here?

Reply:
i end up needing to do this to calculate the right token count